components/Vote/utils/createVoteHandler.ts

```diff
@@ -1,78 +1,99 @@
 import { UPVOTE, DOWNVOTE } from "~/config/constants";
 import API from "~/config/api";
-import { AuthorProfile, UnifiedDocument } from "~/config/types/root_types";
+import {
+  AuthorProfile,
+  CreatedBy,
+  GrmVoteType,
+  ID,
+  RhDocumentType,
+} from "~/config/types/root_types";
+import { SyntheticEvent } from "react";
+import {
+  emptyFncWithMsg,
+  isNullOrUndefined,
+  nullthrows,
+} from "~/config/utils/nullchecks";
+import { buildGrmVoteApiUri } from "~/config/utils/buildGrmVoteApiUri";
 
 type Args = {
-  voteType: string,
-  unifiedDocument: UnifiedDocument,
-  currentAuthor: AuthorProfile,
-  selectedUserVote?: string,
-  onSuccess: Function,
-  onError: Function,
-}
-
-const getVoteUrl = ({ voteType, unifiedDocument }) => {
-  if (unifiedDocument?.documentType === "post") {
-    if (voteType === UPVOTE) {
-      return API.RH_POST_UPVOTE(unifiedDocument.document.id); 
-    }
-    else if (voteType === DOWNVOTE) {
-      return API.RH_POST_DOWNVOTE(unifiedDocument.document.id); 
-    }
-  }
-  else if (unifiedDocument?.documentType === "paper") {
-    if (voteType === UPVOTE) {
-      return API.UPVOTE(unifiedDocument.documentType, unifiedDocument.document.id); 
-    }
-    else if (voteType === DOWNVOTE) {
-      return API.DOWNVOTE(unifiedDocument.documentType, unifiedDocument.document.id); 
-    }
-  }
-}
+  currentAuthor: AuthorProfile;
+  currentVote: GrmVoteType | null | undefined;
+  commentPayload?: {
+    commentID?: string;
+    commentType?: string;
+    replyID?: string;
+    threadID?: string;
+  };
+  documentCreatedBy: any;
+  documentID: ID;
+  documentType: RhDocumentType;
+  onError: Function;
+  onSuccess: ({
+    increment,
+    voteType,
+  }: {
+    increment: number;
+    voteType: GrmVoteType;
+  }) => void;
+  voteType: GrmVoteType;
+};
 
 export const createVoteHandler = ({
-  voteType,
-  unifiedDocument,
+  commentPayload,
   currentAuthor,
-  selectedUserVote,
+  currentVote,
+  documentCreatedBy,
```

Comment:
Can we pass in `DocumentCreatedBy` as `CreatedBy` type ? There is a parser called `parseCreatedBy` that we can use outside of this function

Reply:
I think this is a good thing because we should aim to use types exclusively if possible in newly created components and functions